File: api/src/main/java/com/aistore/module/unitconversion/entity/UnitConversion.java
package com.aistore.module.unitconversion.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/** 计量换算实体（表 unit_conversion） */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("unit_conversion")
public class UnitConversion {

    @TableId(value = "id", type = IdType.AUTO)
    private Long id;
    private Long skuId;
    private String fromUnit;
    private String toUnit;
    private BigDecimal factor;
    private Integer status;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
    private Long createdBy;
    private Long updatedBy;

    @TableLogic
    private Integer deleted;
}

File: web/src/types/unitConversion.ts
/* 计量换算模块类型（对齐 OpenAPI 契约） */
export interface UnitConversion {
  id: number;
  skuId: number;
  skuCode?: string | null;
  skuName?: string | null;
  fromUnit: string;
  toUnit: string;
  factor: number;
  status: 0 | 1;
  createdAt: string;
  updatedAt: string;
  createdBy?: number | null;
  updatedBy?: number | null;
}

export interface UnitConversionSummary {
  id: number;
  skuId: number;
  skuName?: string | null;
  fromUnit: string;
  toUnit: string;
  factor: number;
  status: 0 | 1;
  createdAt: string;
}

export interface UnitConversionListResponse {
  items: UnitConversionSummary[];
  total: number;
  page: number;
  pageSize: number;
  totalPages: number;
}

export interface CreateUnitConversionRequest {
  skuId: number;
  fromUnit: string;
  toUnit: string;
  factor: number;
  status?: 0 | 1;
}

export interface UpdateUnitConversionRequest {
  factor?: number;
}

export interface UnitConversionQueryParams {
  skuId?: number;
  status?: 0 | 1;
  page?: number;
  pageSize?: number;
}
